server/controllers/admin/employees.js

```diff
@@ -0,0 +1,313 @@
+/**
+* The /employees HTTP API endpoint
+*
+* @module categorised/employees
+*
+* @description This controller is responsible for implementing all crud on the
+* employees table through the `/employees` endpoint.
+*
+* @requires lib/db
+*
+* NOTE: This api does not handle the deletion of employees because
+* that subject is not in the actuality.
+*/
+
+'use strict';
+
+var db = require('./../../lib/db');
+
+/**
+* Returns an array of each employee in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // GET /employees : Get list of employees
+* var employees = require('categorised/employees');
+* employees.list(request, response, next);
+*/
+exports.list = function (req, res, next) {
+  var sql =
+    'SELECT ' +
+    'employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
+    'employee.postnom, employee.sexe, employee.dob, employee.date_embauche, employee.service_id, ' +
+    'employee.nb_spouse, employee.nb_enfant, employee.grade_id, employee.locked, grade.text, grade.basic_salary, ' +
+    'fonction.id AS fonction_id, fonction.fonction_txt, ' +
+    'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, employee.daily_salary, employee.location_id, ' +
+    'grade.code AS code_grade, debitor.uuid as debitor_uuid, debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
+    'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+    'FROM employee ' +
+    ' JOIN grade ON employee.grade_id = grade.uuid ' +
+    ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+    ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+    ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+    ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+    ' ORDER BY employee.name ASC, employee.postnom ASC, employee.prenom ASC';
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* Get list of availaible holidays for an employee
+*/
+exports.listHolidays = function (req, res, next) {
+  var pp = JSON.parse(req.params.pp);
+  var sql =
+    'SELECT holiday.id, holiday.label, holiday.dateFrom, holiday.percentage, holiday.dateTo ' +
+    'FROM holiday WHERE ' +
+    '((holiday.dateFrom >= ? AND holiday.dateFrom <= ?) OR ' +
+    '(holiday.dateTo >= ? AND holiday.dateTo <= ?) OR ' +
+    '(holiday.dateFrom <= ? AND holiday.dateTo >a ?)) AND ' +
```

Comment:
I'm pretty sure you didn't write this code as well, but I am just leaving this note ...   This SQL will throw an error.  It should read
```sql
AND holiday.dateTo >= ?
```
but instead reads
```sql
AND holiday.dateTo >a ?
```
If you have more fixes, it would be useful to make this change.